### PR TITLE
Add `ZINTER` command.

### PR DESCRIPTION
**Pull request by Yury-Fridlyand**
https://redis.io/docs/latest/commands/zinter/

copy-paste from https://github.com/aws/glide-for-redis/pull/1338/files

### CODE REVIEW

File: java/client/src/main/java/glide/api/commands/SortedSetBaseCommands.java [hunk 1]
```diff
@@ -1154,6 +1154,128 @@ CompletableFuture<Map<String, Double>> zunionWithScores(
...
+     * <code>keysOrWeightedKeys</code>. To get the elements with their scores, see {@link
+     * #zinterWithScores}.
+     *
+     * @apiNote When in cluster mode, all keys in <code>keysOrWeightedKeys</code> must map to the same
+     *     hash slot.
+     * @since Redis 6.2 and above.
+     * @see <a href="https://redis.io/commands/zinter/">redis.io</a> for more details.
+     * @param keysOrWeightedKeys The keys of the sorted sets with possible formats:
+     *     <ul>
+     *       <li>Use {@link KeyArray} for keys intersection.
```
Comment (aaron-congo) [inline java/client/src/main/java/glide/api/commands/SortedSetBaseCommands.java:1199]:
```suggestion
     *       <li>Use {@link KeyArray} for keys only.
```

File: java/integTest/src/test/java/glide/TransactionTestUtilities.java [hunk 2]
```diff
@@ -380,6 +384,10 @@ private static Object[] sortedSetCommands(BaseTransaction<?> transaction) {
                     Map.of("one", 1.0, "two", 2.0), // zunionWithScores(KeyArray({zSetKey2, zSetKey1}));
                     Map.of("one", 1.0, "two", 2.0), // zunionWithScores(KeyArray({zSetKey2, zSetKey1}), MAX)
                     0L, // zinterstore(zSetKey1, new String[] {zSetKey2, zSetKey1})
+                    0L, // zinter(new KeyArray(new String[] {zSetKey2, zSetKey1}))
+                    0L, // zinter(new KeyArray(new String[] {zSetKey2, zSetKey1}), Aggregate.MAX)
+                    0L, // zinterWithScores(new KeyArray(new String[] {zSetKey2, zSetKey1}))
+                    0L, // zinterWithScores(new KeyArray(new String[] {zSetKey2, zSetKey1}), Aggregate.MAX)
```
Comment (aaron-congo) [inline java/integTest/src/test/java/glide/TransactionTestUtilities.java:390]: Isn't the return type of these transaction commands String[] for zinter and Map for zinterWithScores?

File: java/integTest/src/test/java/glide/SharedCommandTests.java [hunk 3]
```diff
@@ -2479,6 +2479,87 @@ public void zunion(BaseClient client) {
...
+                Map.of("two", -7.0),
+                client
+                        .zinterWithScores(
+                                new WeightedKeys(List.of(Pair.of(key1, 1.0), Pair.of(key2, -2.0))), Aggregate.MIN)
+                        .get());
+
+        // Non-existing Key - empty intersection
+        assertEquals(0, client.zinter(new KeyArray(new String[] {key1, key3})).get().length);
+        assertEquals(0, client.zinterWithScores(new KeyArray(new String[] {key1, key3})).get().size());
+
```
Comment (aaron-congo) [inline java/integTest/src/test/java/glide/SharedCommandTests.java:2553]: Sho has been getting me to add tests that pass an empty key list, and asserting that a RequestException is thrown. Should we add that here? Or is it not possible to construct an empty key list with KeyArray or WeightedKeys?

File: java/integTest/src/test/java/glide/TransactionTestUtilities.java (hunk 2 above)
Comment (Yury-Fridlyand) [reply java/integTest/src/test/java/glide/TransactionTestUtilities.java]: Fixed, a copy-paste issue.

File: java/integTest/src/test/java/glide/SharedCommandTests.java (hunk 3 above)
Comment (Yury-Fridlyand) [reply java/integTest/src/test/java/glide/SharedCommandTests.java:2553]: Added, thank you

**Comment by Yury-Fridlyand** [comment]
Please, ignore node CI failures - fixed in https://github.com/aws/glide-for-redis/pull/1424

**Review by aaron-congo — APPROVED** [review]
